lib/public/AppFramework/Db/BaseEntity.php

```diff
@@ -0,0 +1,263 @@
+	 * getter method
+	 * @since 7.0.0
+	 */
+	public function __call($methodName, $args){
+		$attr = lcfirst( substr($methodName, 3) );
+
+		if(strpos($methodName, 'set') === 0){
+			$this->setter($attr, $args);
+		} elseif(strpos($methodName, 'get') === 0) {
+			return $this->getter($attr);
+		} else {
+			throw new \BadFunctionCallException($methodName .
+				' does not exist');
+		}
+
+	}
+
+
+	/**
+	 * Mark am attribute as updated
+	 * @param string $attribute the name of the attribute
+	 * @since 7.0.0
+	 */
+	protected function markFieldUpdated($attribute){
+		$this->_updatedFields[$attribute] = true;
+	}
+
+
+	/**
+	 * Transform a database columnname to a property
+	 * @param string $columnName the name of the column
+	 * @return string the property name
+	 * @since 7.0.0
+	 */
+	public function columnToProperty($columnName){
+		$parts = explode('_', $columnName);
+		$property = null;
+
+		foreach($parts as $part){
+			if($property === null){
+				$property = $part;
+			} else {
+				$property .= ucfirst($part);
+			}
+		}
+
+		return $property;
+	}
+
+
+	/**
+	 * Transform a property to a database column name
+	 * @param string $property the name of the property
+	 * @return string the column name
+	 * @since 7.0.0
+	 */
+	public function propertyToColumn($property){
+		$parts = preg_split('/(?=[A-Z])/', $property);
+		$column = null;
+
+		foreach($parts as $part){
+			if($column === null){
+				$column = $part;
+			} else {
+				$column .= '_' . lcfirst($part);
+			}
+		}
+
+		return $column;
+	}
+
+
+	/**
+	 * @return array array of updated fields for update query
+	 * @since 7.0.0
+	 */
+	public function getUpdatedFields(){
+		return $this->_updatedFields;
+	}
+
+
+	/**
+	 * Adds type information for a field so that its automatically casted to
+	 * that value once its being returned from the database
+	 * @param string $fieldName the name of the attribute
+	 * @param string $type the type which will be used to call settype()
+	 * @since 7.0.0
+	 */
+	protected function addType(string $fieldName, string $type){
+		$this->_fieldTypes[$fieldName] = $type;
+	}
+
+
+	/**
+	 * Slugify the value of a given attribute
+	 * Warning: This doesn't result in a unique value
+	 * @param string $attributeName the name of the attribute, which value should be slugified
+	 * @return string slugified value
+	 * @since 7.0.0
+	 */
+	public function slugify(string $attributeName): string {
+		// toSlug should only work for existing attributes
+		if(property_exists($this, $attributeName)){
+			$value = $this->$attributeName;
+			// replace everything except alphanumeric with a single '-'
+			$value = preg_replace('/[^A-Za-z0-9]+/', '-', $value);
+			$value = strtolower($value);
+			// trim '-'
+			return trim($value, '-');
+		} else {
+			throw new \BadFunctionCallException($attributeName .
+				' is not a valid attribute');
+		}
+	}
+
+	/**
+	 * @since 18.0.0
+	 */
+	public function convertToType(string $name, $value) {
```

Comment:
```suggestion
	protected function convertToType(string $name, $value) {
```

Seems to be only used in the base and sub class

Reply:
Ah right dark magic ;)